packages/api/core/src/api/init-scripts/init-link.ts

```diff
@@ -0,0 +1,38 @@
+import path from 'path';
+
+import { safeYarnOrNpm, yarnOrNpmSpawn } from '@electron-forge/core-utils';
+import { ForgeListrTask } from '@electron-forge/shared-types';
+import debug from 'debug';
+
+import { readRawPackageJson } from '../../util/read-package-json';
+
+const d = debug('electron-forge:init:link');
+
+/**
+ * Link local forge dependencies
+ *
+ * This allows developers working on forge itself to easily init
+ * a local template and have it use their local plugins / core / cli packages.
+ *
+ * Note: `yarn link:prepare` needs to run first before dependencies can be
+ * linked.
+ */
+export async function initLink<T>(dir: string, task?: ForgeListrTask<T>) {
+  const shouldLink = process.env.LINK_FORGE_DEPENDENCIES_ON_INIT;
+  if (shouldLink) {
+    d('Linking forge dependencies');
+    const packageJson = await readRawPackageJson(dir);
+    const packageManager = safeYarnOrNpm();
+    const linkFolder = path.resolve(__dirname, '..', '..', '..', '..', '..', '..', '.links');
+    for (const packageName of Object.keys(packageJson.devDependencies)) {
+      if (packageName.startsWith('@electron-forge/')) {
+        if (task) task.output = `${packageManager} link --link-folder ${linkFolder} ${packageName}`;
+        await yarnOrNpmSpawn(['link', '--link-folder', linkFolder, packageName], {
```

Comment:
This is where it was silently failing in our tests (found after wrapping it in a `try` `catch`):

<img width="709" alt="Screen Shot 2023-04-23 at 4 11 42 pm" src="https://user-images.githubusercontent.com/1256477/233869913-e39c8a5b-018b-4ef6-b3c3-113521cc5287.png">